spec/models/population_spec.rb

```diff
@@ -0,0 +1,176 @@
+    end
+
+    it 'excludes moves on a different date' do
+      location = create(:location, :prison)
+      create(:move, :prison_transfer, date: Date.tomorrow, to_location: location)
+      population = create(:population, date: Date.today, location: location)
+
+      expect(population.moves_to).to be_empty
+    end
+
+    it 'excludes moves from a different location' do
+      location = create(:location, :prison)
+      create(:move, :prison_transfer, date: Date.today)
+      population = create(:population, date: Date.today, location: location)
+
+      expect(population.moves_to).to be_empty
+    end
+  end
+
+  describe '#free_spaces' do
+    it 'subtracts unavailable space from usable capacity' do
+      population = create(:population, usable_capacity: 10, unlock: 1, bedwatch: 1, overnights_in: 1, overnights_out: 0, out_of_area_courts: 0, discharges: 0)
+      expect(population.free_spaces).to eq(7)
+    end
+
+    it 'adds available space to usable capacity' do
+      population = create(:population, usable_capacity: 10, unlock: 0, bedwatch: 0, overnights_in: 0, overnights_out: 1, out_of_area_courts: 1, discharges: 1)
+      expect(population.free_spaces).to eq(13)
+    end
+
+    it 'subtracts transfers in from usable capacity' do
+      move = create(:move, :prison_transfer)
+      population = create(:population, location: move.to_location, date: move.date, usable_capacity: 10, unlock: 0, bedwatch: 0, overnights_in: 0, overnights_out: 0, out_of_area_courts: 0, discharges: 0)
+      expect(population.free_spaces).to eq(9)
+    end
+
+    it 'adds transfers out to usable capacity' do
+      move = create(:move, :prison_transfer)
+      population = create(:population, location: move.from_location, date: move.date, usable_capacity: 10, unlock: 0, bedwatch: 0, overnights_in: 0, overnights_out: 0, out_of_area_courts: 0, discharges: 0)
+      expect(population.free_spaces).to eq(11)
+    end
+  end
+
+  describe '.free_spaces_date_range' do
+    let!(:prison1) { create(:location, :prison) }
+    let!(:prison2) { create(:location, :prison) }
+    let!(:move1) { create(:move, :prison_transfer, from_location: prison1, date: Date.today) }
+    let!(:move2) { create(:move, :prison_transfer, to_location: prison2, date: Date.today) }
+    let!(:move3) { create(:move, :prison_transfer, from_location: prison1, date: Date.tomorrow) }
+    let!(:population1) { create(:population, location: prison1, date: Date.today) }
+    let!(:population2) { create(:population, location: prison2, date: Date.today) }
+    let!(:population3) { create(:population, location: prison1, date: Date.tomorrow) }
+    let!(:population4) { create(:population, location: prison2, date: Date.tomorrow) }
+    let(:date_range) { (Date.yesterday..Date.tomorrow) }
+    let(:locations) { Location.where(id: [prison1.id, prison2.id]) }
+
+    let(:expected_hash) do
+      {
+        prison1.id => [
+          nil,
+          {
+            free_spaces: population1.free_spaces,
```

Comment:
This is where `free_spaces` method is used at present, and will also be called in the forthcoming `GET /populations/:id` endpoint.